src/components/Facility/FacilityUsers.tsx

```diff
@@ -32,10 +33,22 @@ export default function FacilityUsers(props: { facilityId: number }) {
   if (userListLoading) {
     return <div>Loading...</div>;
   }
+
   if (!userListData) {
     return <div>{t("no_users_found")}</div>;
   }
 
+  const filteredUsers = searchTerm
+    ? userListData.results.filter((user) => {
+        const searchString = searchTerm.toLowerCase();
+        return (
+          user.username?.toLowerCase().includes(searchString) ||
+          user.first_name?.toLowerCase().includes(searchString) ||
+          user.last_name?.toLowerCase().includes(searchString)
+        );
+      })
+    : userListData.results;
+
```

Comment:
Assume user page has 100 users, your api will only give you 1st 14, this search will only on that 14!!!